### ISSUE TITLE
Tree: still keeps scrolling when and after dragging tree items 

### ISSUE THREAD

**Issue by gu-fan** [godotengine/godot#60422]
### Godot version

3.4.4.stable.mono

### System information

MacOS 11.6.5

### Issue description

In Tree, when the scrolling bar is showing. and also is dragging a treeitem.
The scrolling bar will also moving while dragging, and will continue moving after dragging is over.
should have another click on tree to stop it.

furthermore, I have found no doucment for how to control or disable the scrolling behavior.

### Steps to reproduce

just create a scene with tree and attach this script
````
extends Tree
signal moved(item, to_item, shift)

func _ready():

        set_hide_root(true)

        var root = create_item()
        var child1 = create_item(root)
        var child2 = create_item(root)
	for i in 20:
		var subchild1 = create_item(child1)
		subchild1.set_text(0, "Subchild1")





func get_drag_data(position): # begin drag
	var selected = get_selected()
	if !selected: return
	
        set_drop_mode_flags(DROP_MODE_INBETWEEN | DROP_MODE_ON_ITEM)

	# preview
	var cl = ColorRect.new()
	cl.color = Color('#6640FF')
	cl.rect_size = Vector2(40,15)
	var preview = Label.new()
	preview.text = selected.get_text(0)
	preview.rect_position = Vector2(8,2)
	cl.add_child(preview)
	set_drag_preview(cl)

	return selected


func can_drop_data(position, data):
	return data is TreeItem # you shall not pass!


func drop_data(position, item): # end drag

	var to_item = get_item_at_position(position)
	var shift = get_drop_section_at_position(position)
	emit_signal('moved', item, to_item, shift)

````

**Comment by XDRiVE888** [comment]
I confirm this bug also in the ScrollContainer in GdScript. If the ScrollContainer contains elements with the get_drag_data function active, then scrolling after clicking continues to follow the mouse pointer even outside the element. This bug makes it almost impossible to use drag and drop inside a ScrollContainer.

**Comment by gu-fan** [comment]
so this bug may be issued with the scroll container, you may create another issue

**Comment by XDRiVE888** [comment]
> so this bug may be issued with the scroll container, you may create another issue

Yes, I will probably do so if there is no reaction from the developers here. I think that this is a related bug, the fact is that when the dragging of an element is triggered when it is clicked, then the signal InputEventMouseButton.pressed = true is sent to the element and its parents(if mouse_filter = MOUSE_FILTER_PASS), but when the button is released, the signal is not sent, while without the implementation of drag and drop, the mouse release signal is always sent to the element. This is easy to check with the _gui_input(InputEvent event) function callback.

**Comment by nihiluis** [comment]
I think this bug is still occuring on 4.5?

**Comment by fyrddotdev** [comment]
even in 4.6.3 stable
